goaci.go

```diff
@@ -38,45 +41,75 @@ func debug(i ...interface{}) {
 	}
 }
 
+type StringVector []string
+
+func (v *StringVector) String() string {
+	return "\"" + strings.Join(*v, "\" \"") + "\""
+}
+
+func (v *StringVector) Set(str string) error {
+	*v = append(*v, str)
+	return nil
+}
+
 func main() {
+	var (
+		execOpts            StringVector
+		goDefaultBinaryDesc string
+		goBinaryOpt         string
+		goPathOpt           string
+	)
+
+	// Find the go binary
+	gocmd, err := exec.LookPath("go")
+
+	flag.Var(&execOpts, "exec", "Parameters passed to app, can be used multiple times")
+	if err != nil {
+		goDefaultBinaryDesc = "Go binary to use (default: none found in $PATH, so it must be provided)"
+	} else {
+		goDefaultBinaryDesc = "Go binary to use (default: whatever go in $PATH)"
+	}
+	flag.StringVar(&goBinaryOpt, "go-binary", gocmd, goDefaultBinaryDesc)
+	flag.StringVar(&goPathOpt, "go-path", "", "Custom GOPATH (default: a temporary directory)")
+	flag.Parse()
 	if os.Getenv("GOPATH") != "" {
-		die("to avoid confusion GOPATH must not be set")
+		warn("GOPATH envvar is ignored, use --go-path=\"$GOPATH\" option instead")
```

Comment:
Oh, and as of GOPATH - you usually have it set in your environment, but I think that usually you don't want to override it in `goaci` - clean and reproducible builds should be by default.